### PR TITLE
PLF-8765: Oauth integration: sign up/in with linkedin account iss…

### PR DESCRIPTION
**Pull request by lbengayed**
…ue migrate to Oauth2
 ON May 1, 2019, LinkedIn deprecate the use of its v1 APIs.
Applications requesting Version 1.0 APIs experience issues as services begin to remove.
we need to migrate to Version 2.0 of our APIs and OAuth 2.0.

### CODE REVIEW

File: component/web/oauth-common/src/main/java/org/gatein/security/oauth/linkedin/LinkedinProcessorImpl.java [hunk 1]
```diff
@@ -46,9 +51,12 @@
     private final String redirectURL;
     private final String apiKey;
     private final String apiSecret;
+    private final String scope="r_liteprofile r_emailaddress w_member_social";
```
Comment (ahamdi) [inline component/web/oauth-common/src/main/java/org/gatein/security/oauth/linkedin/LinkedinProcessorImpl.java]:
```suggestion
    private final String scope = "r_liteprofile r_emailaddress w_member_social";
```

File: component/web/oauth-web/src/main/java/org/gatein/security/oauth/web/OAuthProviderFilter.java [hunk 2]
```diff
@@ -109,7 +110,7 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
                 }
                 interactionState = getOauthProviderProcessor().processOAuthInteraction(httpRequest, httpResponse, scopeToUse);
             }
-        } catch (OAuthException ex) {
+        } catch (OAuthException | ExecutionException | InterruptedException ex) {
             log.warn("Error during OAuth flow with: " + ex.getMessage());
```
Comment (ahamdi) [inline component/web/oauth-web/src/main/java/org/gatein/security/oauth/web/OAuthProviderFilter.java]: Change it to error and print the exception please

File: pom.xml [hunk 3]
```diff
@@ -73,6 +73,8 @@
     <npm.version>6.1.0</npm.version>
 
     <!-- to move to gatein-dep -->
+    <version.scribejava-apis>6.9.0</version.scribejava-apis>
+    <version.scribejava-core>6.9.0</version.scribejava-core>
     <version.scribe>1.3.5</version.scribe>
```
Comment (ahamdi) [inline pom.xml]: to remove this one if replaced by newer version.

**Review by ahamdi — CHANGES_REQUESTED** [review]
- Please add a unit test
 - Please check that during the build, the right version of scribe library is used

File: pom.xml (hunk 3 above)
Comment (lbengayed) [reply pom.xml]: it used to import Token and OAuthService to verify if the token null or no [https://github.com/Meeds-io/gatein-portal/pull/14/files#diff-4b6c1978f7bb2d0fad7bf290fe443d51R114](url)

File: component/web/oauth-common/src/main/java/org/gatein/security/oauth/linkedin/LinkedinProcessorImpl.java [hunk 4]
```diff
@@ -158,8 +157,8 @@ public LinkedinAccessTokenContext validateTokenAndUpdateScopes(LinkedinAccessTok
 
     @Override
     public void saveAccessTokenAttributesToUserProfile(UserProfile userProfile, OAuthCodec codec, LinkedinAccessTokenContext accessToken) {
-        String encodedAccessToken = codec.encodeString(accessToken.accessToken.getToken());
-        String encodedAccessTokenSecret = codec.encodeString(accessToken.accessToken.getSecret());
+        String encodedAccessToken = codec.encodeString(accessToken.accessToken.getAccessToken());
+        String encodedAccessTokenSecret = codec.encodeString(accessToken.accessToken.getAccessToken());
```
Comment (ahamdi) [inline component/web/oauth-common/src/main/java/org/gatein/security/oauth/linkedin/LinkedinProcessorImpl.java]: What is the difference between those variables ? aren't they the same ?

File: component/web/oauth-common/src/test/java/org/gatein/security/oauth/test/TestOAuthPrincipalProcessor.java [hunk 5]
```diff
@@ -105,4 +116,29 @@ public void testLinkedInGenerateGateInUser() {
         assertEquals("lastName", user.getLastName());
         assertEquals("displayName", user.getDisplayName());
     }
+
+    public void testprocessOAuthInteraction() throws IOException, ExecutionException, InterruptedException {
+
```
Comment (ahamdi) [inline component/web/oauth-common/src/test/java/org/gatein/security/oauth/test/TestOAuthPrincipalProcessor.java:120]: This tests the Scribe Java API, we need to test LinkedinProcessorImpl.java instead

**Review by ahamdi — CHANGES_REQUESTED** [review]

**Review by ahamdi — APPROVED** [review]
LGTM

**Pull request merged**